src/extension.ts

```diff
@@ -30,4 +36,16 @@ export function activate(context: vscode.ExtensionContext) {
 }
 
 export function deactivate() {
+}
+
+function isWsl(terminalSettings) {
+    const windowsShellPath = terminalSettings.integrated.shell.windows;
+
+    if(!windowsShellPath) {
+        return false;
+    }
+
+    const lowerCase = windowsShellPath.toLowerCase();
+
+    return (lowerCase.indexOf('system32') !== -1) && (lowerCase.indexOf('bash') !== -1);
```

Comment:
We could probably make this detection a little better by checking if Windows is running in WoW64 mode ([more info](http://www.growingwiththeweb.com/2017/08/windows-wow64.html)) and also using `%windir%`, you can see how this is done in VS Code here:

https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/terminal/electron-browser/terminalService.ts#L172-L185

Reply:
Agree. Seeing the reference, I feel that there's no common module that detects bash used in the target system. But in this case, I can implement the detection in the same way, and refer to the source code from `src/extension.ts`. Let me make another commit for it.